src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -114,7 +114,11 @@ private void CheckAbstractClassImplementations(BindingDiagnosticBag diagnostics)
             var forwardingMethods = ArrayBuilder<SynthesizedExplicitImplementationForwardingMethod>.GetInstance();
             var methodImpls = ArrayBuilder<(MethodSymbol Body, MethodSymbol Implemented)>.GetInstance();
 
-            // NOTE: We can't iterator over this collection directly, since it is not ordered.  Instead we
+            // A source symbol can have the same interface specified multiple times with different nullability annotations.
+            // This set is used to ensure only distinct interface method implementations are synthesized.
+            var implementedMethods = new HashSet<MethodSymbol>(SymbolEqualityComparer.IgnoringNullable);
```

Comment:
>SymbolEqualityComparer.IgnoringNullable [](http://example.com/codeflow?start=63&length=39)

I assume the same problem could be caused by dynamic differences, tuple names differences, and who knows what differences might be added in the future. Please add tests for differences that I specifically mentioned.
Since this is about metadata, we should be using metadata rules, i.e. ```SymbolEqualityComparer.CLRSignature```. This is also going to be consistent with how ```CalculateInterfacesToEmit``` detects duplicates

Reply:
I'd tested some of those and they were errors, but worth adding tests

Reply:
The mentioned differences are errors, already tested:
- `InheritanceBindingTests.DynamicMismatch_01`
- `CodeGenTupleTests.DuplicateInterfaceDetectionWithDifferentTupleNames_01`